chainer/functions/math/sparse_matmul.py

```diff
@@ -13,8 +13,8 @@
     _scipy_available = False
 
 
-def _coo_matmul(sp_data, sp_row, sp_col, sp_shape, dn, transa, transb, transc,
-                dtype=None):
+def _coo_matmul(sp_data, sp_row, sp_col, sp_shape, sp_row_major,
```

Comment:
How about using the term `c_contiguous` to align with numpy (i.e. `np.arange(...).flags.c_contiguous` ?

Reply:
Thanks for your suggestion. I'd like to use one that is easier to understand for python/numpy developers. Regarding an argument of class `CooMatrix`, would you think it is better to use `c_contiguous` instead of `row_major` for that as well?

Reply:
I think so, basically replacing all occurrences of `row_major` with `c_contiguous` ?

Reply:
Assuming they actually mean the same thing of course. I have not understood the implementation details of this optimization but I also understood that you cannot simply look at the memory layout of the given `ndarray` ?

Reply:
Thank you! I did not think about checking if it is c_contiguous or f_contiguous from the given ndarray. Actually it is probably possible with O(N) complexity.

Reply:
So maybe we should reconsider the name, loop up how e.g. scipy does it. There's also reference from CuPy! https://docs-cupy.chainer.org/en/stable/reference/generated/cupy.sparse.csr_matrix.html#cupy.sparse.csr_matrix.toarray Some suggestions that has been brought up already are `c_order`.

Reply:
I went through scipy code (https://github.com/scipy/scipy/blob/v1.1.0/scipy/sparse/coo.py). Actually there is an argument named `order` in `def toarray(...)`, though it is about order of a dense matrix to be created from a coo sparse matrix.  Anyway, it seems to me that `order='C'/'F'/None` or `c_order=True/False/None` is good. Either would be fine to me, but my preference is rather `c_order`.

Reply:
Could you explain why you'd prefer `c_order` ? I think it's good to follow scipy (and cupy). Then we don't have to "introduce" a new concept.

Reply:
I have no strong reason on that. I just did not think `c_order` was a really new concept. Anyway, I'm OK to go with `order`.

Reply:
Sorry, maybe not a new concept. Neither do I have a very strong opinion. I think the main points are: 
- Will the default behavior be "None" / "no assumption"? 
- Are there possibilities for other orders except C and F?

Reply:
I do think though that `order=None` looks more clear than `c_order=None`.

Reply:
I agree with you on that `order=None` looks more clear.

Re: possibilities for other orders, I don't think we need to consider other orders except **'C'** and **'F'** as long as it is a matrix (2D array). BTW, this is not order, but it may be better to add kind of **'auto'** to the options, and the order is automatically checked by looking at row and column index arrays only when **'auto'** is explicitly specified by users. What would you think?